05.ls/ls.rb

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require 'optparse'
+
+MAX_COLUMN = 3
+SPACE_LENGTH = 3
+
+def main
+  files = search_files(ARGV[0])
+  output(files)
+end
+
+def search_files(path)
+  return [path] if FileTest.file?(path.to_s)
+
+  dirpath = ARGV.empty? ? Dir.getwd : path
+  files = []
+  Dir.foreach(dirpath) do |file|
+    files << file unless file.start_with?('.')
+  end
```

Comment:
ちょっと高等テクニックですが、こんなふうにも書けるはずですー

```suggestion
  files = Dir.foreach(dirpath).reject do |file|
    file.start_with?('.')
  end
```

Reply:
foreachメソッドにさらにメソッド追加するというのは思いつかなかったです！アドバイス通り修正しました。